mpqp/noise/noise_model.py

```diff
@@ -0,0 +1,288 @@
+                    f"Dimension of Depolarizing is {dimension}, but got specified gate(s) of different size."
+                )
+
+        nb_targets = len(targets)
+        if nb_targets < dimension:
+            raise ValueError(
+                f"Number of target qubits {nb_targets} should be higher than the dimension {dimension}. "
+            )
+
+        super().__init__(targets, gates)
+        self.proba = prob
+        """Probability, or error rate, of the depolarizing noise model."""
+        self.dimension = dimension
+        """Dimension of the depolarizing noise model."""
+
+    def to_kraus_representation(self):
+        """3M-TODO"""
+        # generate Kraus operators for depolarizing noise
+        kraus_operators = [...]  # list of Kraus operators
+        return KrausRepresentation(kraus_operators)
+
+    def __repr__(self):
+        return (
+            f"{type(self).__name__}({self.proba}, {self.targets}, {self.dimension}"
+            + (", " + str(self.gates) if self.gates else "")
+            + ")"
+        )
+
+    def to_other_language(
+        self, language: Language = Language.QISKIT
+    ) -> BraketNoise | TwoQubitDepolarizing | QLMNoise:
+        """See documentation of this method in abstract mother class :class:`NoiseModel`.
+
+        Args:
+            language: Enum representing the target language.
+
+        Examples:
+            >>> braket_depo = Depolarizing(0.3, [0,1], dimension=1).to_other_language(Language.BRAKET)
+            >>> braket_depo
+            Depolarizing('probability': 0.3, 'qubit_count': 1)
+            >>> type(braket_depo)
+            braket.circuits.noises.Depolarizing
+            >>> qlm_depo = Depolarizing(0.3, [0,1], dimension=1).to_other_language(Language.MY_QLM)
+            >>> print(qlm_depo)
+            Depolarizing channel, p = 0.3:
+            [[0.83666003 0.        ]
+             [0.         0.83666003]]
+            [[0.        +0.j 0.31622777+0.j]
+             [0.31622777+0.j 0.        +0.j]]
+            [[0.+0.j         0.-0.31622777j]
+             [0.+0.31622777j 0.+0.j        ]]
+            [[ 0.31622777+0.j  0.        +0.j]
+             [ 0.        +0.j -0.31622777+0.j]]
+            >>> type(qlm_depo)
+            qat.quops.quantum_channels.QuantumChannelKraus
+
+        """
+        if language == Language.BRAKET:
+            if self.dimension > 2:
+                raise NotImplementedError(f"Depolarizing channel is not implemented in Braket for more than 2 qubits.")
+            elif self.dimension == 2:
+                return TwoQubitDepolarizing(probability=self.proba)
+            else:
+                return BraketDepolarizing(probability=self.proba)
+
+        elif language == Language.MY_QLM:
+            if self.dimension > 2:
+                raise NotImplementedError(f"Depolarizing channel is not implemented in the QLM for more than 2 qubits.")
+            elif self.dimension == 2 and self.gates is None:
```

Comment:
Indeed. We can also replace by ``if not self.gates`` right ?

Reply:
we could, but in these cases I prefer being explicit rather than relying on truthy values. This is one of the cases where I find that writing more is beneficial, because it allows you to see that you are testing specifically for an empty list (and not a `None` value, a `0`, or any of these other falsy values)